src/inlineSuggestions/registerHandlers.ts

```diff
@@ -88,6 +91,14 @@ export default async function registerInlineHandlers(
     window.onDidChangeTextEditorSelection((e) => {
       console.log("onDidChangeTextEditorSelection: ", e.kind);
     });
+    window
+      .getInlineCompletionItemController(inlineCompletionsProvider)
+      .onDidShowCompletionItem((e) => {
+        const shownCompletionIsMultiline = e.completionItem.text.includes("\n");
+        if (shownCompletionIsMultiline) {
+          void setState({ [StatePayload.SNIPPET_SHOWN]: {} });
+        }
+      });
```

Comment:
@dimacodota idk if its cool to use this api:
![image](https://user-images.githubusercontent.com/67855609/149639288-dbc4a872-1771-4e46-ab06-65608908adc4.png)

Reply:
why not use the  `provideInlineCompletionItems`?

Reply:
because there we're only returning the results, there's an entire event loop iteration that happens between that and when the plugin actually shows the hint.. im guessing that if you move the cursor and such it could send false positives